keyring/system.go

```diff
@@ -53,3 +53,10 @@ func (s *systemKeyring) Get(name string) ([]byte, error) {
 func (s *systemKeyring) Delete(user string) error {
 	return keyring.Delete(s.service, user)
 }
+
+func (s *systemKeyring) List() ([]string, error) {
+	// The OS keyring does not support listing keys
+	// This function is a stub for now because the Keyring interface requires it
+	// Currently, the 'defradb keyring list' command uses only fileKeyring
+	return nil, nil
```

Comment:
todo: I was worried you would leave this func like this :) Sorry I should have clarified on discord.  I agree that this is the best choice, however I think it should error, not silently return - otherwise the users will go bald from the large amount of head scratching that they'll do when trying to figure out why their added keys never appear in the keyring! (it looks like it is empty when the call list on an os-keyring no?)

Reply:
I see the logic behind it throwing an error. My original thoughts were that this function should never actually get called so there was no need to have it do anything. But as I think over it some more, that's all the more reason to make it throw an error. I think you are right.

Reply:
I've made the changes. I'm going to tag you for re-review, so let me know what you think!